framework/components/AAutoTheme/AAutoTheme.js

```diff
@@ -0,0 +1,70 @@
+import React, { useCallback, useMemo, useState, useRef } from "react";
+import { useIsomorphicLayoutEffect, useMediaQuery } from "../../utils/hooks";
+import { useATheme } from "../ATheme";
+import AAutoThemeContext from "./AAutoThemeContext"
+
+const persistKey = 'atomic-react-auto-theme';
+
+const AAutoTheme = (props) => {
+    const { children, } = props;
+    const {persist, setCurrentTheme, currentTheme} = useATheme();
+    // When user turns off auto theme, use the previous theme they manually
+    // selected
+    const prevSelectedTheme = useRef(currentTheme);
```

Comment:
What's the reasoning with this being `useRef` vs `useState`? I don't necessarily thing it's the wrong choice, but just trying to reason about it vs why `isEnabled` is using `useState` as they update the values in different ways.

Reply:
I went with `useRef` for setting the previous theme because it shouldn't cause a re-render (since we are just using it for a future render, i.e., when automatic theming becomes disabled). In the case of `isEnabled`, though, that is stored to keep allow a consuming application to determine if the automatic theming is enabled. The reason it needs to be in state is because we check local storage for a key that indicates if the automatic theming is enabled/disabled. We don't get key/value pair from local storage until we are in the DOM, and if that happens, we need to update React state so that consuming applications will get that state update to know if automatic theming is enabled/disabled. If that were a `useRef`, then the hook wouldn't cause a consuming application to re-render, and they would have a state enabled/disabled property from `useAAutoTheme`. If we weren't exposing `isEnabled` to a consumer, then we could keep it in `useRef`.